packages/react-devtools-shared/src/devtools/views/Components/InspectedElement.js

```diff
@@ -222,10 +224,21 @@ export default function InspectedElementWrapper(_: Props) {
       return;
     }
 
-    const url = new URL(editorURL);
-    url.href = url.href.replace('{path}', source.fileName);
-    url.href = url.href.replace('{line}', String(source.lineNumber));
-    window.open(url);
+    let url = editorURL;
+    url = url.replace('{path}', source.fileName);
+    url = url.replace('{line}', String(source.lineNumber));
+    if (url.startsWith('http')) {
+      // fix https://github.com/facebook/react/issues/24795
+      fetch(url);
+    } else {
```

Comment:
Can you explain what is the purpose of this? I don't get the use case of this.

If we will migrate from using `window.open` to `chrome.tabs.create` on every browser, will we still need this?

Reply:
If the URL uses the IDEA or VS Code protocol, then it is necessary to open a new tab to call the local editor to open the corresponding code file. But if the URL uses the HTTP protocol, then at the browser level, it is possible to directly use the native fetch function to seamlessly open the desired content without any interruption, resulting in a better user experience. This is similar to the experience provided by tools like Vue Devtools, which does not require opening a new tab.

Reply:
> Can you explain what is the purpose of this? I don't get the use case of this.
> 
> If we will migrate from using `window.open` to `chrome.tabs.create` on every browser, will we still need this?

If we are migrating from using window.open to chrome.tabs.create on every browser, then this fetch function may not be necessary.

Reply:
Just to fix this issue, which is https://github.com/facebook/react/issues/24795.

Reply:
Thanks for explaining, are you sure that just fetching the resource will be enough? e.g (`fetch('http://localhost:3000/...`)

From what I see, Vue is also using https://github.com/yyx990803/launch-editor under the hood, so although I believe this might provide a better UX, I am not sure if calling a native `fetch` will work properly in every case.

Reply:
> Thanks for explaining, are you sure that just fetching the resource will be enough? e.g `fetch('http://localhost:3000/...`
> 
> From what I see, Vue is also using https://github.com/yyx990803/launch-editor under the hood, so although I believe this might provide a better UX, I am not sure if calling a native `fetch` will work properly in every case.

Do you have any better suggestions? At the time, I searched for examples in the react-devtools repository and found some that used fetch, so I just went ahead and used fetch.

Reply:
Alright, let's use chrome.tabs.create this time and consider using fetch later if necessary.